git-secrets

```diff
@@ -111,18 +111,63 @@ scan_history() {
 git_grep() {
   local options="$1"; shift
   local files=("${@}") combined_patterns=$(load_combined_patterns)
+  local status=0
 
   [ -z "${combined_patterns}" ] && return 1
-  GREP_OPTIONS= LC_ALL=C git grep -nwHEI ${options} "${combined_patterns}" -- "${files[@]}"
+
+  if [ ${#files[@]} -eq 0 ]; then
+    GREP_OPTIONS= LC_ALL=C git grep -nwHEI ${options} "${combined_patterns}"
+    return $?
+  fi
+
+  # let xargs watch for system limit on arg count for us. xargs returns 123 if
+  # any call returned 1, but we care that all calls returned 1, so invert the
+  # output - xargs will return 0 if every call returned 0
+  printf "%s\n" "${files[@]}" |
+      GREP_OPTIONS= LC_ALL=C xargs -P "$(nproc)" -d'\n' sh -c \
+      'git grep "$@"; [ $? -eq 1 ]' - \
+	-nwHEI "${options}" "${combined_patterns}" --
+
+  # uninvert xargs's 0, which means all git grep invocations returned 1.
+  [ "$?" -ne 0 ]
 }
 
 # Performs a regular grep, taking into account patterns and recursion.
 # Note: this function returns 1 on success, 0 on error.
 regular_grep() {
   local files=("${@}") patterns=$(load_patterns) action='skip'
+  local status=0
   [ -z "${patterns}" ] && return 1
   [ ${RECURSIVE} -eq 1 ] && action="recurse"
-  GREP_OPTIONS= LC_ALL=C grep -d "${action}" -nwHEI "${patterns}" "${files[@]}"
+
+  if [ "${files[@]}" = "-" ]; then
```

Comment:
if [[ "${files[#]}" == 1 && "${files[0]}" == "-" ]; then

Reply:
ACK. Didn't see the reply to my comment earlier about this. Thanks.